sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/optimizer/UnwrapCastInBinaryComparisonSuite.scala

```diff
@@ -120,10 +120,10 @@ class UnwrapCastInBinaryComparisonSuite extends PlanTest with ExpressionEvalHelp
   }
 
   test("unwrap cast should skip when expression is non-deterministic or foldable") {
-    Seq(positiveInt, negativeInt).foreach (v => {
-      val e = Cast(First(f, ignoreNulls = true), IntegerType) <=> v
+    Seq(positiveLong, negativeLong).foreach (v => {
+      val e = Cast(SparkPartitionID(), LongType) <=> v
       assertEquivalent(e, e, evaluate = false)
-      val e2 = Cast(Literal(30.toShort), IntegerType) >= v
+      val e2 = Cast(Literal(30), LongType) >= v
       assertEquivalent(e2, e2, evaluate = false)
```

Comment:
There was no other non-deterministic expression, that can return a short, so I had to change this test a bit.